variables.tf

```diff
@@ -73,6 +73,47 @@ variable "cloud_monitoring_instance_crn" {
   default     = null
 }
 
+##############################################################
+# CSPM
+##############################################################
+
+variable "cspm_enabled" {
+  description = "Enable Cloud Security Posture Management (CSPM) for the Workload Protection instance. This will create a trusted profile for the App Config instance and associate it with the Workload Protection instance."
+  type        = bool
+  default     = true
+  nullable    = false
+}
+
+variable "app_config_crn" {
+  description = "The CRN of the App Config instance to use with the Workload Protection instance. Can be `null` if `cspm_enabled` is not enabled. Must be a valid App Config CRN."
+  type        = string
+  default     = null
+  validation {
+    condition     = var.cspm_enabled ? var.app_config_crn != null : true
+    error_message = "Cannot be `null` if CSPM is enabled. Must be a valid App Config CRN."
```

Comment:
I would remove "Must be a valid App Config CRN" from here since this validation is not checking that